cfme/middleware/provider.py

```diff
@@ -174,3 +175,7 @@ def load_details(self, refresh=False):
             tmp_provider = _db_select_query(
                 name=self.name, type='ManageIQ::Providers::Hawkular::MiddlewareManager').first()
             self.db_id = tmp_provider.id
+
+    def load_topology_page(self):
+        self.load_details()
```

Comment:
I would prefer self.summary.reload() in this case.

Using summary implies load_details. But it is more importnat to make sure the summary is fresh and we are on the page which this ensures.